concept/thing/impl/ThingImpl.java

```diff
@@ -120,9 +121,13 @@ public void setHas(Attribute attribute, boolean isInferred) {
         } else if (getType().getOwns(true).findFirst(attribute.getType()).isPresent()) {
             if (getHas(attribute.getType()).first().isPresent()) {
                 throw exception(TypeDBException.of(THING_KEY_OVER, attribute.getType().getLabel(), getType().getLabel()));
-            } else if (attribute.getOwners(getType()).first().isPresent()) {
-                throw exception(TypeDBException.of(THING_KEY_TAKEN, ((AttributeImpl<?>) attribute).getValue(),
-                        attribute.getType().getLabel(), getType().getLabel()));
+            } else {
+                ThingType topLevelOwnerType = getType();
+                while (topLevelOwnerType.getSupertype().getOwns(attribute.getType().getValueType()).hasNext()) topLevelOwnerType = topLevelOwnerType.getSupertype();
+                if (attribute.getOwners(getType()).first().isPresent()) {
```

Comment:
should `getOwners()` use `toplevelOwnerType`?
I'd still do `intersect(attribute.getType().getSuperTypes(), getSuperTypes()).iterator().next()`. Shorter!
Also, the variable could be more briefly named `ownerType` and let the reader figure it out from the RHS :)